Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -118,27 +118,32 @@ export class ArtifactEngine {
         }
     }
 
-    shouldDownload(path: string) {
-        var shouldDownload = false;
+    shouldDownload(path: string): boolean {
+        let matchedPattern;
         for (let pattern of this.patternList) {
-            if (pattern.charAt(0) === '-' && (minimatch(path, pattern.substr(1), { dot: true, nocase: true }))) {
-                return false;
-            }
-            else {
-                if (minimatch(path, (pattern.charAt(0) === '+') ? pattern.substr(1) : pattern, { dot: true, nocase: true })) {
-                    shouldDownload = true;
+            let _pattern = (pattern.startsWith('+') || pattern.startsWith('-')) ? pattern.substr(1) : pattern;
+            if (minimatch(path, _pattern, { dot: true, nocase: true })) {
+                if (!matchedPattern || minimatch(_pattern, ((matchedPattern.startsWith('+') || matchedPattern.startsWith('-')) ? matchedPattern.substr(1) : matchedPattern), { dot: true, nocase: true })) {
+                    matchedPattern = pattern;
                 }
             }
         }
-        return shouldDownload;
+
+        return (!matchedPattern || matchedPattern.startsWith('-')) ? false : true;
     }
 
+
+    //Supported pattern strings:
```

Comment:
we should support these two patterns also:
** (note that there are no quotes - this would be the default)
drop/dir1/** (this would be given if there is only one spec)
"+drop/dir1/**","-drop/dir1/dir2/**" (no space between first & second)

it is not clear if the below logic takes care of these specs. could you pl. confirm? you can also update the comments to include these specs.